lib/test_hook.rb

```diff
@@ -41,8 +43,40 @@ def post_process_file(file, result, status)
   private
 
   def to_examples(examples)
-    defaults = {preconditions: {}}
-    examples.each_with_index.map { |example, index| defaults.merge(example).merge(id: index) }
+    examples.each_with_index.map do |example, index|
+      example[:preconditions] = classify(example.fetch(:preconditions, {}))
+      example.merge(id: index)
+    end
+  end
+
+  def classify(fields)
+    classified_fields = {}
+    fields.map do |key, value|
+      field = key.to_s
+      classified_fields.deep_merge!(records: { key => value }) if record?(field)
```

Comment:
You can avoid this repeated logic by creating a function that takes the key and returns its category. Then, you can do something like:

```ruby
kind = classify field 
unless kind == :unknown
  classified_fields[kind].merge! { key => value }
end
```